data/src/main/java/org/apache/iceberg/data/TableMigrationUtil.java

```diff
@@ -215,11 +250,7 @@ private static DataFile buildDataFile(
         .build();
   }
 
-  private static ExecutorService migrationService(int parallelism) {
-    return MoreExecutors.getExitingExecutorService(
```

Comment:
why are we switching here from an exiting executor service?

Reply:
`ThreadPools.newWorkerPool` also returns an exiting executor service.